registry/backoff.go

```diff
@@ -0,0 +1,108 @@
+package registry
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+)
+
+var (
+	// For testing
+	now = time.Now
```

Comment:
This is actually used in the code below. I assume you're going to do something about that.